module/Company/src/Company/Controller/AdminController.php

```diff
@@ -256,6 +257,70 @@ public function editCategoryAction()
 
         return $vm;
     }
+
+    /**
+     * Action that displays a form for editing a label
+     *
+     *
+     */
+    public function editLabelAction()
+    {
+        // Get useful stuff
+        $companyService = $this->getCompanyService();
+        $labelForm = $companyService->getLabelForm();
+
+        // Get parameter
+        $labelId = $this->params('labelID');
+
+        // Get the specified company
+        $labels = $companyService->getAllLabelsById($labelId);
+
+        // If the company is not found, throw 404
+        if (empty($labels)) {
+            $company = null;
+            return $this->notFoundAction();
+        }
+
+        // Initialize form
+        $labelsDict = [];
+        foreach ($labels as $label) {
+            $labelsDict[$label->getLanguage()] = $label;
+        }
+        $labelForm->bind($labelsDict);
+        $labelForm->setAttribute(
+            'action',
+            $this->url()->fromRoute(
+                'admin_company/editLabel',
+                [
+                    'labelID' => $labelId,
+                ]
+            )
+        );
```

Comment:
This looks like work that should be partially done in the Form and partially in the View. Not in the controller.